R/create_model_out_submit_tmpl.R

```diff
@@ -0,0 +1,155 @@
+#' Create a model output submission file template
+#'
+#' @param hub_con A `⁠<hub_connection`>⁠ class object.
+#' @inheritParams expand_model_out_val_grid
+#' @param remove_empty_cols Logical. If `FALSE` (default) and `required_vals_only = TRUE`,
+#' task IDs or output_type IDs where all values are optional are included as
+#' columns of `NA`s.
+#' If `TRUE`, such columns are excluded from the output.
```

Comment:
- I don't think we would ever want to drop required task ids or the `output_type_id` column?
- in general, it feels like we should be subsetting rows rather than columns here

Reply:
- here we are allowing the user to drop any columns only contain optional values, not required and only when `required_vals_only` is `TRUE` (hence there are really no values to include in such a column. But this is not the default behaviour, which is to include such columns as `NA`s. There's some discussion about this in one of the resolved comments above with a few more details.